src/use-swr.ts

```diff
@@ -778,7 +757,31 @@ function useSWR<Data = any, Error = any>(
   return memoizedState
 }
 
-const SWRConfig = SWRConfigContext.Provider
+function SWRConfig(props: {
+  children: React.ReactElement
+  value: ConfigInterface
+}) {
```

Comment:
use react build-in type for children 
```ts
const SWRConfig: React.FC<{value: ConfigInterface}> = ({ value, children }) => {/** */}
```

Reply:
@promer94  seems I cannot pass the tsc check without declaring these 2 types?
![image](https://user-images.githubusercontent.com/4800338/98446525-f8cdaf00-2158-11eb-93a1-74a31f123c6f.png)

Reply:
This error message comes from eslint. since we are using typescript, this [***react/prop-types***](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prop-types.md) rule is useless here. 

you can disable it in our eslint config by setting ```"react/prop-types": 0```